tcpdump.c

```diff
@@ -0,0 +1,239 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <errno.h>
+#include <net/if.h>
+#include <linux/if_ether.h>
+#include <linux/if_packet.h>
+#include <linux/filter.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <net/if_arp.h>
+#include <pcap/dlt.h>
+
+#include "error.h"
+#include "bpf_pcap.h"
+#include "pcap_file.h"
+
+// TODO: things i'm not sure about:
+// 1. after create_raw_socket should i pass the sock_fd or a pointer to the sock_fd (right now its sock_fd)
```

Comment:
If i understand your question correctly - you should only pass a pointer to the fd if it's an outparam.